tools/stress_mcp.py

```
@@ -0,0 +1,270 @@
+                    return port
+        except Exception:
+            pass
+    return default_port
+
+
+async def read_exact(reader: asyncio.StreamReader, n: int) -> bytes:
+    buf = b""
+    while len(buf) < n:
+        chunk = await reader.read(n - len(buf))
+        if not chunk:
+            raise ConnectionError("Connection closed while reading")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)